app/Listeners/RoleUpdateLogger.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace App\Listeners;
+
+use HMS\Entities\Role;
+use HMS\Entities\RoleUpdate;
+use App\Events\Roles\UserAddedToRole;
+use Doctrine\ORM\EntityManagerInterface;
+use Illuminate\Queue\InteractsWithQueue;
+use App\Events\Roles\UserRemovedFromRole;
+use HMS\Repositories\RoleUpdateRepository;
+use Illuminate\Contracts\Queue\ShouldQueue;
+
+class RoleUpdateLogger implements ShouldQueue
+{
+    use InteractsWithQueue;
+
+    /**
+     * @var RoleUpdateRepository
+     */
+    protected $roleUpdateRepository;
+
+    /**
+     * @var EntityManagerInterface
+     */
+    protected $entityManager;
+
+    /**
+     * Create the event listener.
+     *
+     * @param RoleUpdateRepository $roleUpdateRepository
+     */
+    public function __construct(RoleUpdateRepository $roleUpdateRepository, EntityManagerInterface $entityManager)
+    {
+        $this->roleUpdateRepository = $roleUpdateRepository;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * Handle user added to role events.
+     *
+     * @param  UserAddedToRole $event
+     */
+    public function onUserAddedToRole(UserAddedToRole $event)
+    {
+        if ($event->role->getName() == Role::MEMBER_APPROVAL) {
+            return;
+        }
+
+        $user = $this->entityManager->merge($event->user);
```

Comment:
The thing gets done before the log update entity gets persisted. What are the needs around atomicity?